### PR TITLE
Patches from Freshrss

### PR DESCRIPTION
**Pull request by Alkarex**
Hello,
Favicon is used in FreshRSS. Please find here a few commits reproduced from the previous version of Favicon https://github.com/FreshRSS/FreshRSS/issues/1501
They address problems with HTTP headers (wrong casing), undefined headers, fileinfo extension not available or buggy.

### CODE REVIEW

File: src/Favicon/Favicon.php [hunk 1]
```diff
@@ -311,9 +314,13 @@ private function checkImageMType($url) {
...
-        $fInfo = finfo_open(FILEINFO_MIME_TYPE);
-        $isImage = strpos(finfo_buffer($fInfo, $content), 'image') !== false;
-        finfo_close($fInfo);
+        $isImage = true;
+        try {
+            $fInfo = finfo_open(FILEINFO_MIME_TYPE);
+            $isImage = strpos(finfo_buffer($fInfo, $content), 'image') !== false;
+            finfo_close($fInfo);
+        } catch (Exception $e) {
+        }
```
Comment (ArthurHoaro) [inline src/Favicon/Favicon.php]: I'm not sure that `finfo_*` will ever raise an exception, but at the very least the error should be logged using `error_log`.

File: src/Favicon/Favicon.php [hunk 2]
```diff
@@ -311,9 +314,13 @@ private function checkImageMType($url) {
...
 
-        $fInfo = finfo_open(FILEINFO_MIME_TYPE);
-        $isImage = strpos(finfo_buffer($fInfo, $content), 'image') !== false;
-        finfo_close($fInfo);
+        $isImage = true;
+        try {
+            $fInfo = finfo_open(FILEINFO_MIME_TYPE);
+            $isImage = strpos(finfo_buffer($fInfo, $content), 'image') !== false;
+            finfo_close($fInfo);
+        } catch (Exception $e) {
```
Comment (ArthurHoaro) [inline src/Favicon/Favicon.php]: `\Exception` because the class is out of the namespace.

Comment (Alkarex) [inline src/Favicon/Favicon.php]: Fixed in https://github.com/ArthurHoaro/favicon/pull/5/commits/b4308ffe168f19c54b368808023f5b0bfa1b62aa

File: src/Favicon/Favicon.php (hunk 1 above)
Comment (Alkarex) [inline src/Favicon/Favicon.php]: Addressed in https://github.com/ArthurHoaro/favicon/pull/5/commits/b4308ffe168f19c54b368808023f5b0bfa1b62aa
The errors occur when fileinfo appears to be installed (which is not always the case), with `extension_loaded('fileinfo')` and `function_exists('finfo_open')` returning true, and the finfo_* functions crashing when actually using them https://github.com/FreshRSS/FreshRSS/issues/1461

**Comment by Alkarex** [comment]
https://github.com/FreshRSS/FreshRSS/pull/1502

File: src/Favicon/Favicon.php (hunk 1 above)
Comment (ArthurHoaro) [inline src/Favicon/Favicon.php]: That an odd issue. Anyway, a try/catch doesn't harm.

**Pull request merged**